onnxscript/function_libs/torch_lib/ops/core.py

```diff
@@ -2260,19 +2260,107 @@ def aten_embedding_backward(
     raise NotImplementedError()
 
 
+@torch_op("aten::embedding_bag", trace_only=True)
 def aten_embedding_bag(
-    weight: TensorType,
-    indices: TensorType,
-    offsets: TensorType,
-    scale_grad_by_freq: bool = False,
-    mode: int = 0,
-    sparse: bool = False,
-    per_sample_weights: Optional[TensorType] = None,
+    weight: TFloat,
+    indices: INT64,
+    offsets: INT64 = None,  # Could be None accotding to the doc, go 2d branch
+    scale_grad_by_freq: bool = False,  # pylint: disable=unused-argument
+    mode: int = 0,  # [0,1,2] indicate ["mean", "sum", "max"]
+    sparse: bool = False,  # pylint: disable=unused-argument
+    per_sample_weights: Optional[TFloat] = None,
     include_last_offset: bool = False,
-) -> tuple[TensorType, TensorType, TensorType, TensorType]:
+) -> TFloat:
     """embedding_bag(Tensor weight, Tensor indices, Tensor offsets, bool scale_grad_by_freq=False, int mode=0, bool sparse=False, Tensor? per_sample_weights=None, bool include_last_offset=False) -> (Tensor, Tensor, Tensor, Tensor)"""
 
-    raise NotImplementedError()
+    if per_sample_weights is None:
+        per_sample_weights = op.Expand(op.Constant(value_floats=[1.0]), op.Shape(indices))
+        per_sample_weights = op.CastLike(per_sample_weights, weight)
+
+    if len(indices.shape) == 1:  # 1d
```

Comment:
We can change this to the normal way of computing rank then combine both functions into a script() one.

Reply:
Tried but failed, because ```op.While()``` cannot be inside ```op.If()```.

Reply:
We should fix onnxscript for this

Reply:
@gramalingam do you have an idea on what would be needed to make this work? Thanks!

Reply:
Yes, I guess this is the same as the issue mentioned in https://github.com/microsoft/onnxscript/issues/879 ... will fix it, thanks

Reply:
#925 is merged. Could you merge from main and see if it works now?